pkg/apis/istio/v1beta1/defaults.go

```diff
@@ -140,6 +140,12 @@ func SetDefaults(config *Istio) {
 	if config.Spec.Pilot.EnableProtocolSniffing == nil {
 		config.Spec.Pilot.EnableProtocolSniffing = util.BoolPointer(true)
 	}
+	if config.Spec.Pilot.EnableProtocolSniffingOutbound == nil {
+		config.Spec.Pilot.EnableProtocolSniffingOutbound = config.Spec.Pilot.EnableProtocolSniffing
+	}
+	if config.Spec.Pilot.EnableProtocolSniffingInbound == nil {
+		config.Spec.Pilot.EnableProtocolSniffingInbound = config.Spec.Pilot.EnableProtocolSniffing
```

Comment:
This seems to be `false` by default: https://github.com/istio/istio/blob/6d0953f791f76f614c2478f463eef166ea399105/pilot/pkg/features/pilot.go#L183

Also, we might consider adding the description from the link to the variables like: `If enabled, protocol sniffing will be used for inbound listeners whose port protocol is not specified or unsupported`

Reply:
👍